gapic/src/main/com/google/gapid/settings.proto

```diff
@@ -114,6 +114,7 @@ message Trace {
   string device_serial = 1;
   string device_name = 2;
   string type = 3;
+  bool default_no_angle = 21;
```

Comment:
Does it make more sense to have this as something like:

```
TraceType trace_type = 21; // or default_trace_type
```
using the `TraceType` enum defined in `gapis/service/service.proto` ?

I think it takes less mental reasoning/gymnastics to understand compared to default_no_angle

Reply:
`type` is already that - it's the last treace type used and becomes the default for future traces. However, we need some extra information to keep track of whether the last *frame* profiling trace (which may not have been the previous trace, if a System trace is taken) was ANGLE or not. The name is a bit odd, I agree, because for the settings the default proto value should be the default for new fields, so that people upgrading don't get odd behavior. Since `false` is the default value, we have to invert the field from `default_angle`.

`type` is a string rather than `Service.TraceType`, because this is serialized and needs to remain backwards compatible and this field is actually older than `Service.TraceType`.

Reply:
Instead of `default_no_angle` maybe something like `default_to_vulkan` instead? Just to get rid of the double negatives.

When calling `setDefaultToVulkan(false)` it makes some amount of sense, but the getter isn't the most clear `getDefaultToVulkan()` (not clear that we're getting a bool here)

Would prefer something like `default_frame_profiling_type` which is an enum to either vulkan or ANGLE, so that the setters & getters both make sense. If we use the TraceType enum in service.proto this would be the vulkan trace and should be okay? Thoughts?

Reply:
Changed it to `last_frame_trace_was_vulkan`.

Originally, I had chosen `not_angle` instead of `vulkan`, because `not angle != vulkan` if we get a third alternative, which arguably may not happen.